pydatastructs/linear_data_structures/linked_lists.py

```diff
@@ -736,3 +738,213 @@ def extract(self, index):
         elif index == 0:
             self.tail.next = self.head
         return node
+
+class SkipList(LinkedList):
+    """
+    Represents Skip List
+
+    Parameters
+    ==========
+
+    p: float
+        Probability that node at level i appears in level i+1.
+
+    levels: int
+        Number of levels in the skip list.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import SkipList
+    >>> skip_list = SkipList(0.5, 16)
+    >>> skip_list.insert(2, "2")
+    >>> skip_list.insert(4, "4")
+    >>> skip_list.insert(6, "4")
+    >>> skip_list.insert(4, "5")
+    >>> skip_list.insert(8, "4")
+    >>> skip_list.insert(9, "4")
+    >>> skip_list.delete(4)
+
+    References
+    ==========
+
+    .. [1] https://epaperpress.com/sortsearch/download/skiplist.pdf
+    .. [2] https://github.com/TheAlgorithms/Python/blob/a9f73e318cddf43769083614a3e1f9dab1ec50fc/data_structures/linked_list/skip_list.py
+
+    """
+
+    __slots__ = ['size', 'level', 'p', 'max_level']
+
+    def __new__(cls, p, max_level):
+        obj = LinkedList.__new__(cls)
+        obj.head = SkipListNode("root", None)
+        obj.level = 0
+        obj.p = p
+        obj.max_level = max_level
+        obj.size = 0
+        return obj
+
+    def __str__(self):
+        """
+        :return: Visual representation of SkipList
```

Comment:
Please use numpy doc string guide.

Reply:
ok will do that.

Reply:
Sir please take a look at tests also, and check if its fine with you.